docs/DeveloperGuide.md

```diff
@@ -315,16 +315,40 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
 2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
 3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+4.  When a new version is released, it should be possible to update the most recent version if the user has installed the previous version.
+5.  Installing a new update shall not in any way, modify or erase existing data and value from the previous version, and the new update should be compatible with the data produced earlier within the system.
+6.  Should be able to store notes in English language, and provisions shall be made to support all languages.
+7.  The system should be able to handle notes with large capacity without any performance liability so that users can keep track of each activity without any issues.
+8.  Whenever a change is made to the application, the progress is instantly saved to the system.
```

Comment:
Is this a functional requirement?

Reply:
I think it's not. it relates to the recoverability of data in this case, where if the software goes wrong somewhere, the progress made on the application will still save the data in the system. This is what I was referring to, but we might want to check if this is a functional requirement haha. Thanks for the heads-up though!

Reply:
The user should not lose any data if the system exits prematurely.